From 0ab6d7a50d37cf4454577cb8c333d4b8683aa054 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Wed, 25 Jul 2012 11:20:36 +0000
Subject: sna: Prefer platform backlight devices over firmware

This is in contrast to libbacklight but closer to our original code as
we prefer a known custom backlight controller over the ACPI interfaces.
As only the ACPI interfaces are marked as firmware, and the custom
backlight controllers as platform, we therefore need to prefer platform
backlight devices.

Reported-by: Austin Lund <austin.lund@gmail.com>
References: https://bugs.freedesktop.org/show_bug.cgi?id=52423
Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
---
diff --git a/src/sna/sna_display.c b/src/sna/sna_display.c
index b31f08d..ebecd21 100644
--- a/src/sna/sna_display.c
+++ b/src/sna/sna_display.c
@@ -282,8 +282,8 @@ sna_output_backlight_get_max(xf86OutputPtr output)
 }
 
 enum {
-	FIRMWARE,
 	PLATFORM,
+	FIRMWARE,
 	RAW,
 	NAMED,
 };
--
cgit v0.9.0.2-2-gbebe
From 484b072072c8297a87940c9e32097923f0a77c8f Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Wed, 25 Jul 2012 11:22:47 +0000
Subject: sna: Add 'gmux_backlight' to list of known devices for backwards compatability

Reported-by: Austin Lund <austin.lund@gmail.com>
References: https://bugs.freedesktop.org/show_bug.cgi?id=52423
Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
---
diff --git a/src/sna/sna_display.c b/src/sna/sna_display.c
index ebecd21..a908596 100644
--- a/src/sna/sna_display.c
+++ b/src/sna/sna_display.c
@@ -292,6 +292,7 @@ static void
 sna_output_backlight_init(xf86OutputPtr output)
 {
 	static const char *known_interfaces[] = {
+		"gmux_backlight",
 		"asus-laptop",
 		"asus-nb-wmi",
 		"eeepc",
--
cgit v0.9.0.2-2-gbebe
